docs: add section for permissions background sync (#9192)
sourcegraph/sourcegraph: doc/admin/repo/permissions.md

Finally, **save the configuration**. You're done!

> ## Background permissions sync

Reviewer: ```suggestion
## Background permissions syncing
```

> Starting at 3.14, Sourcegraph supports syncing permissions in the background, it helps better handle repository permissions at scale. Rather than updating a user’s permissions when they log in and potentially blocking them from seeing search results, Sourcegraph now syncs these permissions proactively, and opportunistically refreshing these permissions in a timely manner, as part of a background service.

Reviewer 2: ```suggestion
Starting with 3.14, Sourcegraph supports syncing permissions in the background to better handle repository permissions at scale. Rather than syncing a user's permissions when they log in and potentially blocking them from seeing search results, Sourcegraph syncs these permissions asynchronously in the background, opportunistically refreshing them in a timely manner.
```

> Background permissions sync is currently behind a feature flag in the [site configuration](../config/site_config.md):

Reviewer: ```suggestion
Background permissions syncing is currently behind a feature flag in the [site configuration](../config/site_config.md):
```

> It is currently behind a feature flag in the [site configuration](../config/site_config.md):

Reviewer 2: ```suggestion
Background permissions sync is currently behind a feature flag in the [site configuration](../config/site_config.md):
```

```json
"permissions.backgroundSync": {
	"enabled": true
}
```

> >NOTE: Only GitLab and Bitbucket Server are supported at this time, GitHub and other code hosts are coming soon.

Author: ```suggestion
>NOTE: Only GitLab and Bitbucket Server are supported at this time, GitHub is coming soon in 3.15.
```

Reviewer: 👍

> >NOTE: Only GitLab and Bitbucket Server are supported at this time, GitHub and other code hosts are coming soon.

Reviewer 2: ```suggestion
>NOTE: Only GitLab and Bitbucket Server are supported at this time. Support for GitHub and other code hosts is coming soon.
```

Author: "other code hosts" is plural right, "is" or "are"? I'm confused ...

Reviewer: We can also say it's coming in 3.15.

> There are some advantages to sync permissions in the background:

Reviewer: ```suggestion
Background permissions syncing has the following benefits:
```

> 1. It schedules the syncing, thus it has more predictable API consumption to the code host.

Reviewer 2: ```suggestion
1. It keeps of schedule of when to sync permissions, resulting in a more predictable load on the code host API.
```

> 1. It keeps of schedule of when to sync permissions, resulting in a more predictable load on the code host API.

Reviewer: ```suggestion
1. More predictable load on the code host API due to maintaining a schedule of permission updates.
```

> 1. It proactively syncs permissions as soon as new repositories are added to the Sourcegraph instance.

Reviewer: ```suggestion
1. Permissions are quickly synced for new repositories added to the Sourcegraph instance.
```

> 1. For users who haven't yet created their accounts on the Sourcegraph instance, they are able to instantly get search results from the repositories they have access on the code host after sign up.

Reviewer 2: ```suggestion
1. Users who sign up on the Sourcegraph instance can immediately get search results from the repositories they have access to on the code host.
```

> However, there are also few things to expect:

Reviewer 2: ```suggestion
Since the syncing of permissions happens in the background, there are a few things to keep in mind:
```

> 1. While kicking off the the initial round of syncing for all repositories and users, users are able to gradually see search results from more repositories they have access to.

Reviewer 2: ```suggestion
1. While the initial sync for all repositories and users is happening, users can gradually see more and more search results from repositories they have access to.
```

> 1. It takes time to complete the first sync. Depending on how many private repositories and users you have on the Sourcegraph instance, it can take from few minutes to several hours. This is generally not a problem for fresh installations, but for existing installations, users may not see the repositories they expect in search results because the permissions syncing hasn't finished yet.

Reviewer: ```suggestion
1. It takes time to complete the first sync. Depending on how many private repositories and users you have on the Sourcegraph instance, it can take from a few minutes to several hours. This is generally not a problem for fresh installations, since admins should only make the instance available after it's ready, but for existing installations, active users may not see the repositories they expect in search results because the initial permissions syncing hasn't finished yet.
```

> 1. It takes time to complete the initial round of syncing. Depending on how many private repositoreis and users you have on the Sourcegraph instance, it could take from few minutes to several hours. This is generally not a problem for fresh installations, but for existing installations, users may not see expected repositories to appear in the search results because the permissions syncing hasn't finished yet.

Reviewer 2: ```suggestion
1. It takes time to complete the first sync. Depending on how many private repositories and users you have on the Sourcegraph instance, it can take from few minutes to several hours. This is generally not a problem for fresh installations, but for existing installations, users may not see the repositories they expect in search results because the permissions syncing hasn't finished yet.
```

> 1. Although we have rate limiting mechanism in place, it consumes more API requests (i.e. more pressure on the code host) during the initial round of syncing.

Reviewer: ```suggestion
1. More requests to the code host API need to be done during the first sync, but their pace is controlled with rate limiting.
```

> Please contact Sourcegraph support if you have any concerns/questions about enabling this feature for your Sourcegraph instance.

Reviewer 2: Can we link to a page that explains how to us contact here?

Author: I thought about just put support@sourcegraph.com, cc @dadlerj for convention here.
